core/domain/suggestion_registry.py

```diff
@@ -352,67 +370,88 @@ def validate(self):
 
         if self.get_score_type() != suggestion_models.SCORE_TYPE_QUESTION:
             raise utils.ValidationError(
-                'Expected the first part of score_category to be %s '
-                ', received %s' % (
+                'Expected the first part of score_category to be "%s" '
+                ', received "%s"' % (
                     suggestion_models.SCORE_TYPE_QUESTION,
                     self.get_score_type()))
-        if not isinstance(self.change_cmd, question_domain.QuestionChange):
+        if not isinstance(self.change, question_domain.QuestionChange):
             raise utils.ValidationError(
-                'Expected change_cmd to be an instance of QuestionChange')
+                'Expected change to be an instance of QuestionChange')
 
-        if not self.change_cmd.cmd:
-            raise utils.ValidationError('Expected change_cmd to contain cmd')
+        if not self.change.cmd:
+            raise utils.ValidationError('Expected change to contain cmd')
 
         if (
-                self.change_cmd.cmd !=
-                question_domain.CMD_ADD_QUESTION_FROM_SUGGESTION):
+                self.change.cmd !=
+                question_domain.CMD_CREATE_NEW_FULLY_SPECIFIED_QUESTION):
             raise utils.ValidationError('Expected cmd to be %s, obtained %s' % (
-                question_domain.CMD_ADD_QUESTION_FROM_SUGGESTION,
-                self.change_cmd.cmd))
+                question_domain.CMD_CREATE_NEW_FULLY_SPECIFIED_QUESTION,
+                self.change.cmd))
 
-        if not self.change_cmd.question_dict:
+        if not self.change.question_dict:
             raise utils.ValidationError(
-                'Expected change_cmd to contain question_dict')
+                'Expected change to contain question_dict')
 
-        if 'question_state_data' not in self.change_cmd.question_dict:
+        if 'question_state_data' not in self.change.question_dict:
```

Comment:
Perhaps I'm missing it, but aren't we supposed to run full validation on this subdict?

Reply:
Hmmm... this was slightly complicated. So the dict doesn't have an ID and a version until the last step where it gets accepted. I have done a question.validate() in the accept function due to this limitation. Is this ok? Else, we might have to replicate the validation (except the ID and version validation) from the question_domain.

Reply:
Done